lib/library.go

```diff
@@ -210,139 +212,59 @@ func Logout() *C.char {
 	return makeJSONResponse(err)
 }
 
-//ApproveSignRequestWithArgs instructs backend to complete sending of a given transaction.
-// gas and gasPrice will be overrided with the given values before signing the
-// transaction.
-//export ApproveSignRequestWithArgs
-func ApproveSignRequestWithArgs(id, password *C.char, gas, gasPrice C.longlong) *C.char {
-	result := statusBackend.ApproveSignRequestWithArgs(C.GoString(id), C.GoString(password), int64(gas), int64(gasPrice))
-
-	return prepareApproveSignRequestResponse(result, id)
-}
-
-//ApproveSignRequest instructs backend to complete sending of a given transaction.
-//export ApproveSignRequest
-func ApproveSignRequest(id, password *C.char) *C.char {
-	result := statusBackend.ApproveSignRequest(C.GoString(id), C.GoString(password))
-
-	return prepareApproveSignRequestResponse(result, id)
-}
-
-// prepareApproveSignRequestResponse based on a sign.Result prepares the binding
-// response.
-func prepareApproveSignRequestResponse(result sign.Result, id *C.char) *C.char {
-	errString := ""
-	if result.Error != nil {
-		fmt.Fprintln(os.Stderr, result.Error)
-		errString = result.Error.Error()
-	}
-
-	out := SignRequestResult{
-		ID:    C.GoString(id),
-		Hash:  result.Response.Hex(),
-		Error: errString,
-	}
-	outBytes, err := json.Marshal(out)
+// SignMessage unmarshals rpc params {data, address, password} and passes
+// them onto backend.SignMessage
+// nolint: deadcode
+func SignMessage(rpcParams *C.char) *C.char {
+	params, err := personal.UnmarshalSignRPCParams(C.GoString(rpcParams))
 	if err != nil {
-		logger.Error("failed to marshal ApproveSignRequest output", "error", err)
-		return makeJSONResponse(err)
+		result := sign.NewErrResult(err)
+		return prepareSignResponse(result)
 	}
-
-	return C.CString(string(outBytes))
+	result := statusBackend.SignMessage(params, params.Password)
+	return prepareSignResponse(result)
 }
 
-//ApproveSignRequests instructs backend to complete sending of multiple transactions
-//export ApproveSignRequests
-func ApproveSignRequests(ids, password *C.char) *C.char {
-	out := SignRequestsResult{}
-	out.Results = make(map[string]SignRequestResult)
-
-	parsedIDs, err := ParseJSONArray(C.GoString(ids))
+// Recover unmarshals rpc params {signDataString, signedData} and passes
+// them onto backend.
+// nolint: deadcode
+func Recover(rpcParams *C.char) *C.char {
```

Comment:
Similar case here.

Reply:
Updated for `signMessage` but recover doesn't take a password.